src/freenet/config/WrapperConfig.java

```diff
@@ -127,10 +136,10 @@ public static synchronized boolean setWrapperProperty(String name, String value)
 		}
 		
 		if(!newConfig.renameTo(oldConfig)) {
-			File oldOldConfig = new File("wrapper.conf.old");
+			File oldOldConfig = new File(wrapperDir + "/wrapper.conf.old");
 			if(oldOldConfig.exists() && !oldOldConfig.delete())
 				try {
-					oldOldConfig = File.createTempFile("wrapper.conf", ".old.tmp", new File("."));
+					oldOldConfig = File.createTempFile(wrapperDir + "wrapper.conf", ".old.tmp", new File("."));
```

Comment:
Thoughts on passing wrapperDir instead of "." for `directory` and not concatenating here?

Reply:
Because then I'd have to re-do the testing done above and duplicate even more code in order to support wrapper being possibly in two locations.

Of course I just now noticed I missed the / on line 142. should be   wrapperDir + "/wrapper.conf"  or  wrapperDir, "wrapper.conf"